sqlmesh/dbt/adapter.py

```diff
@@ -309,13 +317,38 @@ def resolve_identifier(self, relation: BaseRelation) -> t.Optional[str]:
         return identifier
 
     def _map_table_name(
-        self, database: t.Optional[str], schema: t.Optional[str], identifier: t.Optional[str]
+        self,
+        database: t.Optional[str | exp.Identifier],
+        schema: t.Optional[str | exp.Identifier],
+        identifier: t.Optional[str | exp.Identifier],
     ) -> exp.Table:
-        name = ".".join(p for p in (database, schema, identifier) if p is not None)
+        name = ".".join(
```

Comment:
you could do `nullsafe_join(".", database, schema, identifier)` here.

Reply:
don't use null safe join

do 

table_name = exp.table_(identefier, db=schema, catalog=database)

Reply:
nullsafe_join is going to be deprecated soon, it's only used in a test

Reply:
Ah right thanks, I forgot about `table_`

Reply:
I think there might actually be an issue here because if we create a table and then generate SQL from it there can be quotes included. Is there a utility to just generate the name part without quotes?